dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -97,6 +98,25 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
 
     if (try_init_hw_decoder &&
         nvjpegCreate(NVJPEG_BACKEND_HARDWARE, NULL, &handle_) == NVJPEG_STATUS_SUCCESS) {
+    // disable HW decoder for drivers < 455.x as the memory pool for it is not available
```

Comment:
I wonder if it makes sense to try to initialize HW decoder and than switch to regular NVJPEG handle as you still go through all of the config.

Isn't it better to do it in the check above - if we have hw_decoder_load specified, but the driver is old, just set the load to 0 and try_init_hw_decoder to false and don't bother with the setup here.

Reply:
I wanted to show a warning that there is a HW decoder but it is disabled. The best way is to just call `nvjpegCreate(NVJPEG_BACKEND_HARDWARE, NULL, &handle_)` to make sure the HW decoder is available.

Reply:
Anyway, I forgot to skip the rest of the code if the driver is not sufficient.